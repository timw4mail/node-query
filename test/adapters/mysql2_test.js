'use strict';

const configFile = (process.env.TRAVIS) ? '../config-travis.json' : '../config.json';

// Load the test base
const reload = require('require-reload')(require);
reload.emptyCache();
const testBase = reload('../base');
const expect =  testBase.expect;
const promiseTestRunner = testBase.promiseTestRunner;
const testRunner = testBase.testRunner;

// Load the test config file
let adapterName = 'mysql2';
let config = reload(configFile)[adapterName];

// Set up the query builder object
let nodeQuery = reload('../../lib/NodeQuery')(config);
let qb = nodeQuery.getQuery();

suite('Mysql2 adapter tests -', () => {

	suiteSetup(() => qb.truncate('create_test'));

	test('nodeQuery.getQuery = nodeQuery.init', () => {
		expect(nodeQuery.getQuery())
			.to.be.deep.equal(qb);
	});

	//--------------------------------------------------------------------------
	// Callback Tests
	//--------------------------------------------------------------------------
	testRunner(qb, (err, result, done) => {
		expect(err).is.not.ok;
		expect(result.rows).is.an('array');
		expect(result.columns).is.an('array');
		expect(result.rowCount()).to.not.be.undefined;
		expect(result.columnCount()).to.not.be.undefined;
		done();
	});
	test('Callback - Select with function and argument in WHERE clause', done => {
		qb.select('id')
			.from('create_test')
			.where('id', 'CEILING(SQRT(88))')
			.get((err, rows) => {
				expect(err).is.not.ok;
				return done();
			});
	});
	test('Callback - Test Insert Batch', done => {
		let data = [
			{
				id: 5441,
				key: 3,
				val: new Buffer('7'),
			}, {
				id: 891,
				key: 34,
				val: new Buffer('10 o\'clock'),
			}, {
				id: 481,
				key: 403,
				val: new Buffer('97'),
			},
		];

		qb.insertBatch('create_test', data, (err, res) => {
			expect(err).is.not.ok;
			return done(err);
		});
	});

	//---------------------------------------------------------------------------
	// Promise Tests
	//---------------------------------------------------------------------------
	promiseTestRunner(qb);
	test('Promise - Select with function and argument in WHERE clause', () => {
		let promise = qb.select('id')
			.from('create_test')
			.where('id', 'CEILING(SQRT(88))')
			.get();

		return expect(promise).to.be.fulfilled;
	});

	test('Test Insert Batch', () => {
		let data = [
			{
				id: 5442,
				key: 4,
				val: new Buffer('7'),
			}, {
				id: 892,
				key: 35,
				val: new Buffer('10 o\'clock'),
			}, {
				id: 482,
				key: 404,
				val: 97,
			},
		];

		return expect(qb.insertBatch('create_test', data)).to.be.fulfilled;
	});

	suiteTeardown(() => {
		qb.end();
	});
});
