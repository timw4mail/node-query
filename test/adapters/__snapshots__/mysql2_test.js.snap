// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mysql2 adapter tests - Get compiled -  getCompiledDelete 1`] = `"DELETE FROM \`create_test\` WHERE \`id\` = ?"`;

exports[`Mysql2 adapter tests - Get compiled -  getCompiledInsert 1`] = `"INSERT INTO \`create_test\` (\`id\`,\`key\`,\`val\`) VALUES (?,?,?)"`;

exports[`Mysql2 adapter tests - Get compiled -  getCompiledSelect 1`] = `"SELECT \`id\`, \`key\` AS \`k\`, \`val\` FROM \`create_test\` WHERE (\`id\` > ? AND \`id\` < ?) LIMIT 1,2"`;

exports[`Mysql2 adapter tests - Get compiled -  getCompiledSelect 2 1`] = `"SELECT \`id\`, \`key\` AS \`k\`, \`val\` FROM \`create_test\` WHERE (\`id\` > ? AND \`id\` < ?) LIMIT 1,2"`;

exports[`Mysql2 adapter tests - Get compiled -  getCompiledUpdate 1`] = `"UPDATE \`create_test\` SET \`id\`=?,\`key\`=?,\`val\`=? WHERE \`id\` = ?"`;
